protocol/routing/README.md

```diff
@@ -0,0 +1,74 @@
+IPFS Routing Protocol Spec
+==========================
+
+Authors: David Dias
+
+Reviewers:
+
+TODOS:
+
+-----------------------
+
+> This spec defines the routing protocol spec. Routing offers an interface for the features exposed by `Peer discovery` and `DHT`. The spec is a **Work In Progress**.
+
+## Supports
+
+- Routing primitives
+  - Publish and fetch content (also providing)
+- Maintaining partial state of the network
+  - DHT
+  - kbucket
+
+## Overview
+
+The Routing Protocol is composed by three componenets, these are:
+- Interface: Our routing primitives that are offered for the user, such as finding and publishing content, including the storage and lookup of metadata (Providers).
+- Peer Discovery: Responsible for filling our kbucket with best candidates.
```

Comment:
i think discovery should be separate from routing. I should be able to run routing protocols without a single discovery protocol (or a "manual discovery" i.e. i call `ipfs swarm connect <ipfs-addr-of-a-node-i-know`)